faucet.go

```diff
@@ -232,6 +118,50 @@ func cleanAndExpandPath(path string) string {
 	return filepath.Clean(os.ExpandEnv(path))
 }
 
+// Verify timelimit from determined IP and count how many time
+// its need to do a new action.
+func verifyTimeLimit(ip string, actionsTimeLimit time.Duration, mutex sync.RWMutex) error {
+	mutex.RLock()
+	lastRequestTime, found := requestIPs[ip]
+	mutex.RUnlock()
+	if found {
+		nextAllowedRequest := lastRequestTime.Add(actionsTimeLimit)
+		coolDownTime := time.Until(nextAllowedRequest)
+
+		if coolDownTime >= 0 {
+			err := fmt.Errorf("client(%v) may only do an action every "+
+				"%v. Wait another %v.",
+				ip, actionsTimeLimit, coolDownTime)
+			return err
+		}
+	}
+	return nil
+}
+
+// Get the client's real IP address
```

Comment:
Start comment with function name:

> getRealIP returns the clients IP address. If the useRealIP config is set it will use the X-Real-IP or X-Forwarded-For HTTP headers, otherwise it will return the request.RemoteAddr field